src/NServiceBus.Callbacks/ConvertLegacyEnumResponseToLegacyControlMessageBehavior.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using NServiceBus.Features;
+    using NServiceBus.Pipeline;
+    using NServiceBus.Pipeline.Contexts;
+
+    class ConvertLegacyEnumResponseToLegacyControlMessageBehavior : PhysicalOutgoingContextStageBehavior
+    {
+        public override void Invoke(Context context, Action next)
+        {
+            var instance = context.Get<object>("RequestResponse.OutgoingLogicalMessageInstance");
+
+            if (!CallbacksSupport.isLegacyEnumResponse(instance.GetType()))
```

Comment:
I don't see a need to change it.
From my perspective we may not need `context.MessageType` at all